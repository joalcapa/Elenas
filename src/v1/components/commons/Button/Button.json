{
  "description": "Este componente renderiza un boton basico que ejecuta una funcion y se puede deshabilitar",
  "displayName": "Button",
  "methods": [],
  "props": {
    "onPress": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que ejecuta el boton cuando se hace click sobre el.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "text": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Nombre del boton.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "isDefaultButton": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Por defecto el boton es de color azul.",
      "defaultValue": {
        "value": "true",
        "computed": false
      }
    },
    "isGreenButton": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "El boton cambia a color verde.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    },
    "isDisabled": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Se deshabilita el boton.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    }
  }
}