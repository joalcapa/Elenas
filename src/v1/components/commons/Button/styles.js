import {StyleSheet} from 'react-native';

import {Colors, FontSize} from '../../../resources';

const styles = StyleSheet.create({
    button_container: {
        borderRadius: '8px',
        paddingTop: '8px',
        paddingBottom: '8px',
        width: '100%', 
        maxWidth: '150px',
        displayFlex: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    text: {
        fontSize: FontSize.normal,
        color: Colors.C_000000,
    },
    default_button: {
        backgroundColor: Colors.C_4287f5,
    },
    green_button: {
        backgroundColor: Colors.C_42f575,
    },
    disabled_button: {
        backgroundColor: Colors.C_dbdbdb,
    },
});

export default styles;