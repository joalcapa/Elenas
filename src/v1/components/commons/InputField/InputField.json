{
  "description": "Este componente renderiza un input basico.",
  "displayName": "InputField",
  "methods": [],
  "props": {
    "value": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "placeholder": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "onChangeText": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "isPassword": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    }
  }
}