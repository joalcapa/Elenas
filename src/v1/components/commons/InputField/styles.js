import {StyleSheet} from 'react-native';

import {Colors} from '../../../resources';

const styles = StyleSheet.create({
    input: {
        width: '100%',
        maxWidth: 150,
        paddingTop: 5,
        paddingBottom: 5,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5,
        borderWidth: 1,
        borderColor: Colors.C_7e807f,
    },
});

export default styles;