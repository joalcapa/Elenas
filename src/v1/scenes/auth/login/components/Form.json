{
  "description": "Este componente renderiza el formulario para iniciar session.",
  "displayName": "Form",
  "methods": [],
  "props": {
    "cellphone": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del telefono.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "password": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor de la contraseÃ±a.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "isValidForm": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Identifica si el formulario es valido para enviar.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    },
    "isError": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Identifica si ocurrio un error al enviar el formulario.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    },
    "changeCellphone": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que modifica el valor del telefono.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "changePassword": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que modifica la contraseÃ±a.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "login": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que envia el formulario.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    }
  }
}

