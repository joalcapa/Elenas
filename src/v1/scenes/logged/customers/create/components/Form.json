{
  "description": "Este componente permite enviar el formulario para crear o editar un cliente.",
  "displayName": "Form",
  "methods": [],
  "props": {
    "firstName": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del nombre.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "lastName": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del apellido.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "cedula": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor de la cedula.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "email": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del correo.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "cellphone": {
      "type": {
        "name": "string"
      },
      "required": false,
      "description": "Valor del telefono.",
      "defaultValue": {
        "value": "''",
        "computed": false
      }
    },
    "isValidForm": {
      "type": {
        "name": "bool"
      },
      "required": false,
      "description": "Define si el formulario es valido para enviar.",
      "defaultValue": {
        "value": "false",
        "computed": false
      }
    },
    "changeFirstName": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que cambia el valor del nombre.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "changeLastName": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que cambia el valor del apellido.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "changeCedula": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que cambia el valor de la cedula.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "changeEmail": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que cambia el valor del correo.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "changeCellphone": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que cambia el valor del telefono.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    },
    "createCustomer": {
      "type": {
        "name": "func"
      },
      "required": false,
      "description": "Funcion que muestra el formulario de direccion.",
      "defaultValue": {
        "value": "() => {}",
        "computed": false
      }
    }
  }
}
