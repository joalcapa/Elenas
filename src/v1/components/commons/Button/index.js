import React from 'react';
import PropTypes from 'prop-types';
import {TouchableOpacity, Text} from 'react-native';

import styles from './styles';

const Button = (props) => (
        <TouchableOpacity 
            onPress={props.onPress} 
            style={[
                styles.button_container,
                props.isDefaultButton ? styles.default_button : {},
                props.isGreenButton ? styles.green_button : {},
            ]}
        >
            <Text style={styles.text}>{props.text}</Text>
        </TouchableOpacity>
    );

Button.propTypes = {
    onPress: PropTypes.func,
    text: PropTypes.string,
    isDefaultButton: PropTypes.bool,
    isGreenButton: PropTypes.bool,
};

Button.defaultProps = {
    onPress: () => {},
    text: '',
    isDefaultButton: true,
    isGreenButton: false,
};

export default Button;