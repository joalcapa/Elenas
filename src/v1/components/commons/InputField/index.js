import React from 'react';
import PropTypes from 'prop-types';
import {TextInput} from 'react-native';

import styles from './styles';

/**
 * Este componente renderiza un input basico.
 */
const InputField = (props) => 
    <TextInput 
        {...props} 
        style={styles.input}
        secureTextEntry={props.isPassword}
    />;

InputField.propTypes = {
    value: PropTypes.string,
    placeholder: PropTypes.string,
    onChangeText: PropTypes.func,
    isPassword: PropTypes.bool,
};

InputField.defaultProps = {
    /**
     * Valor del input.
     */
    value: '',
    /**
     * Informacion que identifica al input.
     */
    placeholder: '',
    /**
     * Funcion que modifica el valor del input.
     */
    onChangeText: () => {},
    /**
     * Valor que indica si es input de contrase√±a.
     */
    isPassword: false,
};

export default InputField;