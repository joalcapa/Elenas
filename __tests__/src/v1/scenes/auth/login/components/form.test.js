import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {create} from 'react-test-renderer';
import {TouchableOpacity} from 'react-native';
import {shallow, configure} from 'enzyme';

import Form from '../../../../../../../src/v1/scenes/auth/login/components/form.js';
import InputField from '../../../../../../../src/v1/components/commons/InputField';
import Button from '../../../../../../../src/v1/components/commons/Button';

configure({adapter: new Adapter()});
describe('Form component', () => {
    test('render empty form component', () => {
        const tree = create(<Form />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render form component with cellphone', () => {
        const tree = create(<Form cellphone={'+5732485855'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render form component with password', () => {
        const tree = create(<Form password={'123456789'} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render form component with error', () => {
        const tree = create(<Form isError />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onChangeText cellphone functionality', () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(
            <Form changeCellphone={onPressEvent} />,
        );

        wrapper.find(InputField).first().props().onChangeText('value');
        expect(onPressEvent.mock.results[0].value).toBe('Link on press invoked');
    });

    test('onChangeText password functionality', () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(
            <Form changePassword={onPressEvent} />,
        );

        wrapper.find(InputField).at(1).props().onChangeText('value');
        expect(onPressEvent.mock.results[0].value).toBe('Link on press invoked');
    });

    test('onPress functionality', () => {
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(<Form login={onPressEvent} isValidForm />);

        wrapper.find(Button).first().props().onPress();
        expect(onPressEvent.mock.results[0].value).toBe('Link on press invoked');
    });

    test('disabled functionality', () => {
        let isPress = true;
        const onPressEvent = jest.fn();
        onPressEvent.mockReturnValue('Link on press invoked');
        const wrapper = shallow(<Form login={onPressEvent} isValidForm={false} />);

        try {
            wrapper.find(TouchableOpacity).first().props().onPress();
        } catch (error) {
            isPress = false;
        }

        expect(isPress).toBe(false);
    });
});