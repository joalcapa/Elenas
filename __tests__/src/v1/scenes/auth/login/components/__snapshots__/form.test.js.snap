// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component render empty form component 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-height-i7wkem r-justifyContent-1777fci"
>
  <div
    className="css-view-1dbjc4n r-alignContent-uia4a0 r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-justifyContent-1777fci r-maxWidth-z7pwl0 r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-k8qxaj r-paddingTop-1knelpx"
    >
      <div
        className="css-text-901oao r-fontWeight-vw2c0b"
        dir="auto"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Inicio de sesión
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Telefono"
        readOnly={false}
        spellCheck={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Contraseña"
        readOnly={false}
        spellCheck={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-oyv1ms r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
        onClick={[Function]}
        onContextMenu={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "WebkitTransitionDuration": "0s",
            "transitionDuration": "0s",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
          dir="auto"
          onClick={[Function]}
        >
          Iniciar Sesión
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Form component render form component with cellphone 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-height-i7wkem r-justifyContent-1777fci"
>
  <div
    className="css-view-1dbjc4n r-alignContent-uia4a0 r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-justifyContent-1777fci r-maxWidth-z7pwl0 r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-k8qxaj r-paddingTop-1knelpx"
    >
      <div
        className="css-text-901oao r-fontWeight-vw2c0b"
        dir="auto"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Inicio de sesión
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Telefono"
        readOnly={false}
        spellCheck={true}
        type="text"
        value="+5732485855"
      />
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Contraseña"
        readOnly={false}
        spellCheck={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-oyv1ms r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
        onClick={[Function]}
        onContextMenu={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "WebkitTransitionDuration": "0s",
            "transitionDuration": "0s",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
          dir="auto"
          onClick={[Function]}
        >
          Iniciar Sesión
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Form component render form component with error 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-height-i7wkem r-justifyContent-1777fci"
>
  <div
    className="css-view-1dbjc4n r-alignContent-uia4a0 r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-justifyContent-1777fci r-maxWidth-z7pwl0 r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-k8qxaj r-paddingTop-1knelpx"
    >
      <div
        className="css-text-901oao r-fontWeight-vw2c0b"
        dir="auto"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Inicio de sesión
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Telefono"
        readOnly={false}
        spellCheck={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Contraseña"
        readOnly={false}
        spellCheck={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-oyv1ms r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
        onClick={[Function]}
        onContextMenu={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "WebkitTransitionDuration": "0s",
            "transitionDuration": "0s",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
          dir="auto"
          onClick={[Function]}
        >
          Iniciar Sesión
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <div
        className="css-text-901oao r-fontSize-1b43r93"
        dir="auto"
        onClick={[Function]}
      >
        Hubo un error, intenta mas tarde
      </div>
    </div>
  </div>
</div>
`;

exports[`Form component render form component with password 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-height-i7wkem r-justifyContent-1777fci"
>
  <div
    className="css-view-1dbjc4n r-alignContent-uia4a0 r-alignItems-1awozwy r-display-6koalj r-flexDirection-eqz5dr r-justifyContent-1777fci r-maxWidth-z7pwl0 r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-k8qxaj r-paddingTop-1knelpx"
    >
      <div
        className="css-text-901oao r-fontWeight-vw2c0b"
        dir="auto"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Inicio de sesión
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Telefono"
        readOnly={false}
        spellCheck={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="css-view-1dbjc4n r-paddingBottom-1mi0q7o r-paddingTop-m611by"
    >
      <input
        autoCapitalize="sentences"
        autoComplete="on"
        autoCorrect="on"
        className="css-textinput-11aywtz r-borderColor-4yh6xw r-borderRadius-1f0042m r-borderWidth-rs99b7 r-maxWidth-127gp16 r-paddingBottom-1inuy60 r-paddingLeft-1yt7n81 r-paddingRight-ry3cjt r-paddingTop-glunga r-width-13qz1uu"
        dir="auto"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onSelect={[Function]}
        placeholder="Contraseña"
        readOnly={false}
        spellCheck={true}
        type="password"
        value="123456789"
      />
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-justifyContent-1777fci r-paddingBottom-1mi0q7o r-paddingTop-m611by r-width-13qz1uu"
    >
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-oyv1ms r-borderRadius-1xfd6ze r-cursor-1loqt21 r-displayFlex-14n0d7j r-justifyContent-1777fci r-maxWidth-127gp16 r-paddingBottom-xd6kpl r-paddingTop-tskmnb r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr r-width-13qz1uu"
        onClick={[Function]}
        onContextMenu={[Function]}
        onKeyDown={[Function]}
        style={
          Object {
            "WebkitTransitionDuration": "0s",
            "transitionDuration": "0s",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-text-901oao r-color-cqee49 r-fontSize-1b43r93"
          dir="auto"
          onClick={[Function]}
        >
          Iniciar Sesión
        </div>
      </div>
    </div>
  </div>
</div>
`;
